### PR TITLE
feature(avatar-upload):impelement user profile image upload

### PR DESCRIPTION
**Pull request by swaibat**
#### What does this PR do?

- implement user profile picture upload
- add sample configurations in `.env.sample`
- add new packages `cloudinary` and `express-fileUpload`

#### Description of Task to be completed?
nan
#### How should this be manually tested?

- clone the repo
- run npm i 
- npm test

#### Any background context you want to provide?
nan
#### What are the relevant pivotal tracker stories?

[pivotal story](https://www.pivotaltracker.com/story/show/168697552)
#### Questions:

### CODE REVIEW

File: server/index.js [hunk 1]
```diff
@@ -1,10 +1,17 @@
  /* eslint-disable no-console */
 import express from 'express';
+import fileUpload from 'express-fileupload';
 import bodyParser from 'body-parser';
 import { methodError, serverError } from './middlewares/errorsMiddleware';
 import routers from './routes/index';
 
 const app = express();
+
```
Comment (urbanishimwe) [inline server/index.js]: have you tested this because, in my computer this library didn't work well

File: server/routes/authRoute.js [hunk 2]
```diff
@@ -4,8 +4,11 @@ import validateSignup from '../middlewares/singupValidate';
 import signin from '../controllers/signin';
 import validator from '../middlewares/validlogin';
 import checkAdminSignup from '../middlewares/signupAdminMiddleware';
+import decodeToken from '../middlewares/decodeToken';
+import { upload } from '../controllers/requestMaids';
 
 const app = express.Router();
 app.post('/signup', validateSignup, checkAdminSignup, signup, addMaid);
 app.post('/signin', validator, signin);
+app.patch('/users/profile/upload', decodeToken, upload)
```
Comment (urbanishimwe) [inline server/routes/authRoute.js]: check if the user is logged after decodeToken call back

Comment (lemoissonM) [inline server/routes/authRoute.js]: I don't think that a verb **/upload** should be in the route, what it the route is just **/users/:userId/avatar'** or **/users/:userid/profile-picture**

File: server/controllers/requestMaids.js [hunk 3]
```diff
@@ -20,3 +21,12 @@ export async function getSingleReq(req, res) {
 export async function getAllReq(req, res) {
   return sendResult(res, 200, 'Request successfull', req.allReq);
 }
+
```
[the remark this answers is not in the document]
Comment (swaibat) [reply server/controllers/requestMaids.js]: cool rectified

File: server/index.js (hunk 1 above)
Comment (swaibat) [reply server/index.js]: fully tested

File: server/routes/authRoute.js (hunk 2 above)
Comment (urbanishimwe) [reply server/routes/authRoute.js]: why do you need an ID, if the id can be extracted from the token

Comment (swaibat) [reply server/routes/authRoute.js]: let's decide v ia slack how it can be done

Comment (swaibat) [reply server/routes/authRoute.js]: its automatically done id the helper function metre did

File: server/controllers/requestMaids.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (urbanishimwe) [reply server/controllers/requestMaids.js]: so `if (error) sendResult(res, 400, error.message);` can be changed `if (error) return sendResult(res, 400, error.message);`

**Pull request merged**